### ISSUE TITLE
crypto_box: ChaChaBox test failure on 0.9.0-rc.0

### ISSUE THREAD

**Issue by alxiong** [RustCrypto/nacl-compat#124]
I encountered errors when updating to `crypto_box` to `"=0.9.0-rc.0"`:

```toml
# Cargo.toml
[dependencies]
crypto_box = { version = "=0.9.0-rc.0", default-features = false, features = ["chacha20", "rand_core", "alloc"] }
rand_core = { version = "0.6.4", default-features = false }
rand_chacha = { version = "0.3.1", default-features = false }
```
```rust
#[cfg(test)]
mod tests {
    use crypto_box::{
        aead::{Aead, AeadCore, Payload},
        ChaChaBox,
    };
    use rand_core::SeedableRng;
    use rand_chacha::ChaCha20Rng;

    #[test]
    fn test_enc() {
        let mut rng = ChaCha20Rng::from_seed([0u8; 32]);

        let sk = crypto_box::SecretKey::generate(&mut rng);
        let pk = crypto_box::PublicKey::from(&sk);

        let msg = b"The quick brown fox jumps over the lazy dog";
        let aad = b"my associated data";

        let nonce = ChaChaBox::generate_nonce(&mut rng);

        // generate an ephemeral key pair as the virtual sender to derive the crypto box
        let ephemeral_sk = crypto_box::SecretKey::generate(&mut rng);
        let ephemeral_pk = crypto_box::PublicKey::from(&ephemeral_sk);
        let my_box = ChaChaBox::new(&pk, &ephemeral_sk);

        // encrypt the message and associated data using crypto box
        let ct = my_box
            .encrypt(&nonce, Payload { msg, aad })
            .unwrap(); // FAILED HERE, I'M NOT SURE WHY??

        // decrypt
        let recv_box = ChaChaBox::new(&ephemeral_pk, &sk);
        let decrypted = recv_box.decrypt(&nonce, Payload { msg: &ct, aad }).unwrap();
        assert_eq!(decrypted, msg);
    }
}
```
and running `cargo test` produces:
```
running 1 test
test tests::test_enc ... FAILED

failures:

---- tests::test_enc stdout ----
thread 'tests::test_enc' panicked at 'called `Result::unwrap()` on an `Err` value: Error', src/lib.rs:30:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

I wonder if this has to do with the update in https://github.com/RustCrypto/AEADs/pull/295, 
Even if I try prepend with `RUSTFLAGS="--cfg curve25519_dalek_backend="u64_backend" cargo test`, still the same err. 

any idea what I miss? @tarcieri

**Comment by tarcieri** [comment]
AAD is no longer supported by the new construction. Both XSalsa20Poly1305 and XChaCha20Poly1305 are now variants of the [`crypto_secretbox`](https://docs.rs/crypto_secretbox) construction.

This was necessary for compatibility with libsodium. Unfortunately there's nothing we can do about it without having an incompatible construction.

**Comment by alxiong** [comment]
ah, sorry that I miss that info in the doc.

so the recommended way (if I don't care about nacl compatibility) for AAD suport will be using [`(x)chacha20poly1305` ](https://github.com/RustCrypto/AEADs/tree/master/chacha20poly1305), correct?

**Comment by tarcieri** [comment]
Yes, you can extract the old code from v0.8 to use X25519 with IETF-flavored XChaCha20Poly1305

**Comment by tarcieri** [comment]
> could you remind me the reason for removing the aad support? Is it for more compatibility with libsodium?

Yes. The `crypto_secretbox` construction does not support AAD.

**Comment by alxiong** [comment]
@tarcieri do you mind release a new `v0.2.0-rc.0` on `crypto_kx` to depend on `v2.0.0.-*` of `x25519-dalek` (which further depends on `v4.0.0-*` of `curve25519-dalek`?

(Would like to use `curve25519_dalek_backend="u64_backend"` instead of feature flag for backend selection)

Thanks!

**Comment by tarcieri** [comment]
Released in #125
